src/api/email.rs

```diff
@@ -0,0 +1,177 @@
+//! Email service for sending transactional emails (e.g., activation codes)
+
+use lettre::message::header::ContentType;
+use lettre::transport::smtp::authentication::Credentials;
+use lettre::{Message, SmtpTransport, Transport};
+use std::env;
+
+#[derive(Clone)]
+pub struct EmailService {
+    mailer: Option<SmtpTransport>,
+    from_address: String,
+    enabled: bool,
+}
+
+impl EmailService {
+    /// Create a new email service from environment variables
+    ///
+    /// Required environment variables (if email is enabled):
+    /// - `SMTP_USERNAME`: SMTP username
+    /// - `SMTP_PASSWORD`: SMTP password
+    /// - `SMTP_SERVER`: SMTP server address (e.g., smtp.gmail.com)
+    /// - `SMTP_FROM_ADDRESS`: From email address (e.g., noreply@boom.example.com)
+    ///
+    /// Optional:
+    /// - `EMAIL_ENABLED`: Set to "false" to disable email (defaults to true if SMTP vars are set)
+    pub fn new() -> Self {
+        // Check if email should be enabled
+        let enabled = env::var("EMAIL_ENABLED")
+            .unwrap_or_else(|_| "true".to_string())
+            .parse::<bool>()
+            .unwrap_or(true);
+
+        if !enabled {
+            println!("Email service is DISABLED (EMAIL_ENABLED=false)");
+            return Self {
+                mailer: None,
+                from_address: String::new(),
+                enabled: false,
+            };
+        }
+
+        // Try to load SMTP configuration
+        let smtp_username = env::var("SMTP_USERNAME").ok();
+        let smtp_password = env::var("SMTP_PASSWORD").ok();
+        let smtp_server = env::var("SMTP_SERVER").ok();
+        let from_address = env::var("SMTP_FROM_ADDRESS")
+            .unwrap_or_else(|_| "noreply@boom.example.com".to_string());
+
+        // If any SMTP config is missing, disable email
+        if smtp_username.is_none() || smtp_password.is_none() || smtp_server.is_none() {
+            println!(
+                "Email service is DISABLED (missing SMTP configuration: SMTP_USERNAME, SMTP_PASSWORD, or SMTP_SERVER)"
+            );
+            return Self {
+                mailer: None,
+                from_address,
+                enabled: false,
+            };
+        }
+
+        // Build SMTP transport
+        let creds = Credentials::new(smtp_username.unwrap(), smtp_password.unwrap());
+
+        let mailer = match SmtpTransport::relay(&smtp_server.unwrap()) {
+            Ok(transport) => Some(transport.credentials(creds).build()),
+            Err(e) => {
+                eprintln!("Failed to create SMTP transport: {}", e);
+                println!("Email service is DISABLED (SMTP transport creation failed)");
+                return Self {
+                    mailer: None,
+                    from_address,
+                    enabled: false,
+                };
+            }
+        };
+
+        println!("Email service is ENABLED (SMTP configured)");
```

Comment:
we should start using tracing for all of our println. Not a problem of this PR though, I believe someone else is already working on tracing integration